android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -188,66 +187,73 @@ protected void onCreate(Bundle savedInstanceState) {
 
         completeHandler = DownloadCompleteHandler.with(context);
 
-        NetworkUtils.checkConnection(new WeakReference<>(this), new NetworkUtils.NetworkStateReceiverListener() {
-            @Override
-            public void activeConnection() {
-                //Internet is working
-                if (!sharedPreferences.getBoolean(ConstantStrings.IS_DOWNLOAD_DONE, false)) {
-                    DialogFactory.createDownloadDialog(context, R.string.download_assets, R.string.charges_warning, (dialogInterface, button) -> {
-                        if (button==DialogInterface.BUTTON_POSITIVE) {
-                            Boolean preference = sharedPreferences.getBoolean(getResources().getString(R.string.download_mode_key), true);
-                            if (preference) {
-                                disposable.add(NetworkUtils.haveNetworkConnectionObservable(MainActivity.this)
-                                        .subscribeOn(Schedulers.io())
-                                        .observeOn(AndroidSchedulers.mainThread())
-                                        .subscribe(connected -> {
-                                            if (connected) {
-                                                OpenEventApp.postEventOnUIThread(new DataDownloadEvent());
-                                            } else {
-                                                final Snackbar snackbar = Snackbar.make(mainFrame, R.string.internet_preference_warning, Snackbar.LENGTH_INDEFINITE);
-                                                snackbar.setAction(R.string.yes, view -> downloadFromAssets());
-                                                snackbar.show();
-                                            }
-                                        }));
-                            } else {
-                                OpenEventApp.postEventOnUIThread(new DataDownloadEvent());
+        if (Urls.getBaseUrl().equals(Urls.EMPTY_LINK)) {
```

Comment:
@iamareebjamal nesting is required here.
 
Because if what we want is 
if base url empty -> download from assets
else  show dialogue to download from server.

For making code more readable i will shift code for showing dialogue to separate method